test/shared/src/main/scala/zio/test/laws/ZLawsF.scala

```diff
@@ -280,5 +290,14 @@ object ZLawsF {
       final def run[R1 <: R, F[_]: CapsF, A: Caps](genF: GenF[R1, F], gen: Gen[R1, A]): ZIO[R1, Nothing, TestResult] =
         checkM(genF(gen), genF(gen), genF(gen))(apply(_, _, _).map(_.map(_.label(label))))
     }
+
+    /**
+     * Constructs a law from a pure function taking three parameters.
+     */
+    abstract class Law3Function[-CapsF[_[_]], -Caps[_]](label: String) extends Invariant[CapsF, Caps, Any] { self =>
```

Comment:
For invariant, the likely form this law would take would be `<=>` instead of `=>`. So I'm not sure whether to put that in here or to defer it downstream. Thinking about it, nothing stops us from including these type classes in ZIO Prelude, so users can do things like `new zio.prelude.Invaraint.ComposeLaw`.

Reply:
Yes, for this one in particular we have to be quite careful in the way we generate the pair of functions so they form an equivalence relation.